src/api/instance/replace.js

```
@@ -64,6 +87,73 @@ function wait_until_healthy(region, lbName, instanceId) {
 		});
 }
 
+function attach_elastic_ip(region, source_instance_id, alloc_id) {
+	let params = {
+		AllocationId: alloc_id,
+		InstanceId: source_instance_id
+	};
+
+	return AWSProvider
+		.get_ec2(region)
+		.associateAddressAsync(params)
+		.catch(err => {
+			return Promise.reject(new Error(`${region}: Address was not associated to ${source_instance_id}: ${err}`));
+		});
+}
+
+function detach_elastic_ip(region, assoc_id) {
+	let params = {
+		AssociationId: assoc_id
+	};
+
+	return AWSProvider
+		.get_ec2(region)
+		.disassociateAddressAsync(params)
+		.catch(err => {
+			return Promise.reject(new Error(`${region}: Failed to detach elastic IP with association ID ${assoc_id}: ${err}`));
+		});
+}
+
+function get_elastic_ip(region, target_instance_id) {
+	let params = {
+		Filters: [
+			{
+				Name: 'instance-id',
+				Values: [target_instance_id]
+			}
+		]
+	};
+
+	return AWSProvider
+		.get_ec2(region)
+		.describeInstancesAsync(params)
+		.then(data => {
+			if (data && data.Reservations && data.Reservations.length &&
+				data.Reservations[0].Instances && data.Reservations[0].Instances.length) {
+				return data.Reservations[0].Instances[0].PublicIpAddress;
+			}
+		}).then(pub_address => {
+			params = {
+				PublicIps: [pub_address]
+			};
+			return AWSProvider
+				.get_ec2(region)
+				.describeAddressesAsync(params)
+				.then(data => {
+					if (data && data.Addresses && data.Addresses.length) {
+						for (let address of data.Addresses) {
+							if (address.AllocationId && address.AssociationId) {
+								return {
+									alloc_id: address.AllocationId,
+									assoc_id: address.AssociationId
+								};
+							}
+						}
+					}
+				});
+		}).catch((err) => console.log(err));
```

this is also just swallowing an error, which i don't think we want